src/server/activeobjectmgr.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+Minetest
+Copyright (C) 2010-2018 nerzhul, Loic BLOT <loic.blot@unix-experience.fr>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include <log.h>
+#include "mapblock.h"
+#include "profiler.h"
+#include "activeobjectmgr.h"
+
+namespace server
+{
+
+void ActiveObjectMgr::clear(const std::function<bool(ServerActiveObject *, u16)> &cb)
+{
+	std::vector<u16> objects_to_remove;
+	for (auto &it : m_active_objects) {
+		if (cb(it.second, it.first)) {
+			// Id to be removed from m_active_objects
+			objects_to_remove.push_back(it.first);
+		}
+	}
+
+	// Remove references from m_active_objects
+	for (u16 i : objects_to_remove) {
+		m_active_objects.erase(i);
+	}
+}
+
+void ActiveObjectMgr::step(
+		float dtime, const std::function<void(ServerActiveObject *)> &f)
+{
+	g_profiler->avg("Server::ActiveObjectMgr: num of objects",
+			m_active_objects.size());
+	for (auto &ao_it : m_active_objects) {
+		f(ao_it.second);
+	}
+}
+
+bool ActiveObjectMgr::registerObject(ServerActiveObject *obj)
+{
+	assert(obj); // Pre-condition
+	if (obj->getId() == 0) {
+		u16 new_id = getFreeId();
+		if (new_id == 0) {
+			errorstream << "Server::ActiveObjectMgr::addActiveObjectRaw(): "
+				    << "no free id available" << std::endl;
```

Comment:
Please use tabs to indent. (same for the the entire file)

Reply:
clang-format cheat :) i will fix that